src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+import pathlib
+import os
+import yaml
+import json
+import random
+from typing import Literal, List
+import pyarrow as pa
+import pyarrow.parquet as pq
+from fhirpathpy import evaluate
+
+from phdi_building_blocks.fhir import (
+    AzureFhirserverCredentialManager,
+    query_fhir_server,
+    log_fhir_server_error,
+)
+
+
+def load_schema(path: str) -> dict:
+    """
+    Given the path to local YAML files containing a user-defined schema read the file
+    and return the schema as a dictionary.
+
+    :param path: Path specifying the location of a YAML file containing a schema.
+    :return schema: A user-defined schema
+    """
+
+    with open(path, "r") as file:
+        schema = yaml.safe_load(file)
+    return schema
+
+
+def apply_selection_criteria(
+    value: list,
+    selection_criteria: Literal["first", "last", "random", "all"],
+) -> str:
+    """
+    Given a list of value parsed from a FHIR resource and selection criteria return a
+    single value.
+
+    :param value: A list containing the values stored in a top level key in a FHIR
+    resource.
+    :param selection_criteria: A string indicating which element of list to select
+    when one is encountered during parsing.
+    """
+
+    if selection_criteria == "first":
+        value = value[0]
+    elif selection_criteria == "last":
+        value = value[-1]
+    elif selection_criteria == "random":
+        value = random.choice(value)
+    elif selection_criteria == "all":
```

Comment:
This will cause every condition to be checked, but I'll make the change as there aren't many conditions and evaluation is simple.

Reply:
The fact that every condition would be checked is definitely something to keep in mind. As you've stated, there are so few here that I don't think it's an issue. 

However, I did just do some googling for best practices here and found [this SO post](https://stackoverflow.com/questions/22782785/when-to-use-if-vs-elif-in-python), which actually contradicts what I'm saying. They're stating that the use of `elif` demonstrates that the values are mutually exclusive, but several `if` statements don't. Because of this, I've changed my mind and think I was wrong. I actually think we should do it the way you're doing it. I think we can still get rid of the last `elif` statement though since it doesn't actually do anything.